File: bindings/php/src/ProvenDhcp.php
<?php
// SPDX-License-Identifier: PMPL-1.0-or-later
// Copyright (c) 2026 Jonathan D.A. Jewell (hyperpolymath) <j.d.a.jewell@open.ac.uk>
//
// DHCP protocol types for proven-servers.

declare(strict_types=1);

namespace ProvenServers;

/** MessageType matching the Idris2 ABI tags. */
enum MessageType: int
{
    case Discover = 0;
    case Offer = 1;
    case Request = 2;
    case Ack = 3;
    case Nak = 4;
    case Release = 5;
    case Inform = 6;
    case Decline = 7;
}

/** OptionCode matching the Idris2 ABI tags. */
enum OptionCode: int
{
    case SubnetMask = 0;
    case Router = 1;
    case Dns = 2;
    case DomainName = 3;
    case LeaseTime = 4;
    case ServerId = 5;
    case RequestedIp = 6;
    case MsgType = 7;
}

/** HardwareType matching the Idris2 ABI tags. */
enum HardwareType: int
{
    case Ethernet = 0;
    case Ieee802 = 1;
    case Arcnet = 2;
    case FrameRelay = 3;
}

/** DhcpState matching the Idris2 ABI tags. */
enum DhcpState: int
{
    case Idle = 0;
    case DiscoverReceived = 1;
    case OfferSent = 2;
    case RequestReceived = 3;
    case AckSent = 4;
    case NakSent = 5;
}

/** LeaseState matching the Idris2 ABI tags. */
enum LeaseState: int
{
    case Available = 0;
    case Offered = 1;
    case Bound = 2;
    case Renewing = 3;
    case Rebinding = 4;
    case Expired = 5;
}

/** RelaySubOption matching the Idris2 ABI tags. */
enum RelaySubOption: int
{
    case CircuitId = 0;
    case RemoteId = 1;
}

File: bindings/csharp/src/ProvenDhcp.cs
// SPDX-License-Identifier: MPL-2.0
// Copyright (c) 2026 Jonathan D.A. Jewell (hyperpolymath) <j.d.a.jewell@open.ac.uk>
//
// DHCP protocol types for proven-servers.

namespace Proven;

/// <summary>MessageType matching the Idris2 ABI tags (0-7).</summary>
public enum MessageType : byte
{
    Discover = 0,
    Offer = 1,
    Request = 2,
    Ack = 3,
    Nak = 4,
    Release = 5,
    Inform = 6,
    Decline = 7
}

/// <summary>OptionCode matching the Idris2 ABI tags (0-7).</summary>
public enum OptionCode : byte
{
    SubnetMask = 0,
    Router = 1,
    Dns = 2,
    DomainName = 3,
    LeaseTime = 4,
    ServerId = 5,
    RequestedIp = 6,
    MsgType = 7
}

/// <summary>HardwareType matching the Idris2 ABI tags (0-3).</summary>
public enum HardwareType : byte
{
    Ethernet = 0,
    Ieee802 = 1,
    Arcnet = 2,
    FrameRelay = 3
}

/// <summary>DhcpState matching the Idris2 ABI tags (0-5).</summary>
public enum DhcpState : byte
{
    Idle = 0,
    DiscoverReceived = 1,
    OfferSent = 2,
    RequestReceived = 3,
    AckSent = 4,
    NakSent = 5
}

/// <summary>LeaseState matching the Idris2 ABI tags (0-5).</summary>
public enum LeaseState : byte
{
    Available = 0,
    Offered = 1,
    Bound = 2,
    Renewing = 3,
    Rebinding = 4,
    Expired = 5
}

/// <summary>RelaySubOption matching the Idris2 ABI tags (0-1).</summary>
public enum RelaySubOption : byte
{
    CircuitId = 0,
    RemoteId = 1
}
